src/main/perl/Check.pm

```diff
@@ -536,28 +536,28 @@ sub link ($$;%) {
     $target = _realpath($otarget);
     $message = $opt{hard} ? "create hard link $osource => $otarget" :
 	"create symlink $osource -> $otarget";
-    #
-    # check target
-    #
-    if ($otarget =~ /^\//) {
-	$test = $target;
-    } else {
-	$test = $source;
-	$test =~ s=[^/]+$==;
-	$test .= $otarget;
-    }
+
     if ($opt{hard}) {
-	# note: we follow symlinks for $target to be compatible with ln(1)
-	@stat = LC::Fatal::stat($test);
+        # Check target that is required to exist.
+	@stat = LC::Fatal::lstat($target);
 	if (@stat) {
-	    if (-d _) {
+	    if (-d _ && ! -l _) {
```

Comment:
with lstat and the magic `_`, it's either `-d` or `-l`, so the `&& ! -l` is not required
```
[stdweird@gast039e tmp]$ perl -e '@stat=lstat("broken");print "D", -d _, "F", -f _, "E", -e _, "L", -l _,"\n"'
DFE1L1
[stdweird@gast039e tmp]$ perl -e '@stat=lstat("dir");print "D", -d _, "F", -f _, "E", -e _, "L", -l _,"\n"'
D1FE1L
```
without `_` magic, it's 
```
[stdweird@gast039e tmp]$ perl -e 'print "D", -d "broken", "F", -f "broken", "E", -e "broken", "L", -l "broken","\n"'
DFEL1
```

best to add comment 😄